common/gamelib/player.mjs

```diff
@@ -57,11 +85,15 @@ class Player {
       return false;
     }
     // TODO: check whether msg is in the correct format
-    if (msg.mapCoord !== undefined) this.mapCoord = msg.mapCoord;
+    if (msg.mapCoord !== undefined) {
+      this._previousMapCoord = (this.mapCoord ?? msg.mapCoord); // for smoothly move the player
+      this.mapCoord = msg.mapCoord;
+    }
     if (msg.facing !== undefined) this.facing = msg.facing;
     if (msg.displayName !== undefined) this.displayName = msg.displayName;
     if (msg.displayChar !== undefined) this.displayChar = msg.displayChar;
     if (msg.removed !== undefined) this.removed = msg.removed;
+    this.lastMovingTime = Date.now();
```

Comment:
Maybe set `this.lastMovingTime` only if `this.mapCoord` changed? Considering `updateFromMessage()` is not limited to only moving the user.

Reply:
fixed